### DISCUSSION TITLE
Question/Enhancement: Publish to self-hosted minio

### DISCUSSION THREAD

**Discussion by darkdragn** [oclif/oclif#1285] [Q&A]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------

Question/feature: Is it possible to publish to minio, which is s3 compatible?

What is the current behavior?
-----------------------------
Currently, the only reference I can find to the s3 configuration is [in the releasing documentation.](https://oclif.io/docs/releasing)

I was wondering if it is possible to publish to minio, through configuration in package.json. If the package is using aws sdk, this can be achieved with the env var AWS_S3_ENDPOINT. While on the topic, would it be possible to either:

- add dotenv to enable an end user to store AWS_{ACCESS_KEY_ID,SECRET_ACCESS_KEY} in a common location, while auto-populating .gitignore with .env to allow publishing to be more repeatable, without having to define those env vars for each publish. 
- or potentially allowing the publish method to use aws configuration (~/.aws/*) via aws sdk, which would have both keys and the s3 endpoint. This last option would solve all problems mentioned in this issue.

##OFF TOPIC## If possible, since I couldn't find it, could I get a concise breakdown of all fields allowed in the oclif key in package.json? So far, I've seen one or two references scattered through out the documentation and I would love a complete breakdown of what configuration options are allowed in a single location. Instead of playing hunt and find.

What is the expected behavior?
------------------------------
The ability to consistently publish to an s3 compatible hosted service, preferably with a static configuration location to make repeat publishes more simple. (oclif/1.18.1 linux-x64 node-v14.15.4)

**Comment by darkeagle1337** [comment]
I have done it. I used the min.io version: RELEASE.2023-07-21T21-12-44Z (go1.19.11 linux/amd64)

You need to set some environment variables before you can publish the tarballs.

```
export AWS_SECRET_ACCESS_KEY=XXX 
export AWS_Access_Key_ID=YYYYY 
export AWS_S3_ENDPOINT=http://{minio-host}:{minio.port}
export AWS_S3_FORCE_PATH_STYLE=true 
```

Then you can publish the tarballs

`oclif upload tarballs --targets=darwin-arm64`

I found the environment in the code: https://github.com/oclif/oclif/blob/main/src/aws.ts#L58-59

Maybe you the developer can add this to the documentation.
